### ISSUE TITLE
bug of UNet final step dim

### ISSUE THREAD

**Issue by flyfyyfyy** [lucidrains/denoising-diffusion-pytorch#329]
When we use to `init_dim` in UNet model, if it is not equal to `dim`, the model will cause wrong.

The resaon is the self.final_res_block use the wrong `dim`.

To fix it, change blow part in Unit class
```
        self.final_res_block = resnet_block(dim * 2, dim)
        self.final_conv = nn.Conv1d(dim, self.out_dim, 1)
```
to 
```
        self.final_res_block = resnet_block(init_dim * 2, init_dim)
        self.final_conv = nn.Conv1d(init_dim, self.out_dim, 1)
```

Hope you could fix this bug soon.

**Comment by lucidrains** [comment]
@flyfyyfyy oops, fixed!
